app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,41 @@
 class PostsController < ApplicationController
+  before_action :set_users, only: %i[index show new]
+
   def index
     @users = User.find(params[:user_id])
     @posts = @users.posts
   end
 
   def show
     @users = User.find(params[:user_id])
-    @post = @users.posts.find(params[:id])
+    @posts = @users.posts.find(params[:id])
+  end
+
+  def new
+    @users = User.find(params[:user_id])
+    @posts = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
```

Comment:
- [ ] Please make sure all code is written properly for now you declare method `posts_params` but you call `post_params` which gives this error: undefined local variable or method `post_params' for #<PostsController:0x00000000020ad0> so kindly fix this error as well

```suggestion
    @post = current_user.posts.new(posts_params)
```

![Screenshot from 2024-01-28 00-04-27](https://github.com/supreeti/blogapp/assets/45226848/665c78de-2228-46d6-97d2-9a8f827bfb11)


- [ ] After fixing all the errors when I try to create a new post it doesn't create a new post so make sure your all functionalities working as expected :+1: 

- [ ] I didn't check the like and comment functionalities because I am unable to create a new post so kindly check your all functionalities before submitting for re-review :+1: